File: Python/Compare Version Numbers.py
class Solution:
    # @param a, a string
    # @param b, a string
    # @return a boolean
    def compareVersion(self, version1, version2):
        seq1 = []
        seq2 = []
        if version1.find('.') >= 0:
            seq1 = version1.split('.')
        else:
            seq1.append(version1)
        if version2.find('.') >= 0:
            seq2 = version2.split('.')
        else:
            seq2.append(version2)
        for i in range(len(seq1)):
            seq1[i] = int(seq1[i])
        for i in range(len(seq2)):
            seq2[i] = int(seq2[i])
        maxlen = max(len(seq1), len(seq2))
        for i in range(len(seq1), maxlen):
            seq1.append(0)
        for i in range(len(seq2), maxlen):
            seq2.append(0)
        if seq1 < seq2:
            return -1
        elif seq1 > seq2:
            return 1
        else:
            return 0

File: C++/Compare-Version-Numbers.cpp
class Solution {
public:
    int compareVersion(string version1, string version2) {
        int num1 = 0, num2 = 0, len1 = version1.size(), len2 = version2.size();
        int p1 = 0, p2 = 0;
        while (p1 < len1 || p2 < len2) {
        	while (p1 < len1 && version1[p1] != '.') {
        		num1 = num1 * 10 + version1[p1 ++] - '0';
        	}
        	while (p2 < len2 && version2[p2] != '.') {
        		num2 = num2 * 10 + version2[p2 ++] - '0';
        	}
        	if (num1 > num2) return 1;
        	else if (num1 < num2) return -1;
        	num1 = 0;
        	num2 = 0;
        	p1 ++;
        	p2 ++;
        }
        return 0;
    }
};